tools/a-star/src/astar/A_Star.java

```diff
@@ -0,0 +1,136 @@
+package astar;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class A_Star {
+
+
+    public static void linkNeighbors(Spot[][] grid) {
+        for (int i = 0; i < grid.length; i++) {
+            for (int j = 0; j < (grid[i].length) - 1; j++) {
+                if (i > 0) {
+                    grid[i][j].addNeighbor(grid[i - 1][j]);
+                }
+                if (j > 0) {
+                    grid[i][j].addNeighbor(grid[i][j - 1]);
+                }
+                if (i < grid.length - 1) {
+                    grid[i][j].addNeighbor(grid[i + 1][j]);
+                }
+                if (j < grid.length - 1) {
+                    grid[i][j].addNeighbor(grid[i][j + 1]);
+                }
+                if (i > 0 && j > 0) {
+                    grid[i][j].addNeighbor(grid[i - 1][j - 1]);
+                }
+                if (i < grid.length - 1 && j < grid.length - 1) {
+                    grid[i][j].addNeighbor(grid[i + 1][j + 1]);
+                }
```

Comment:
i think the map will look more intuitive and overall just easier too work with without these two movements

Reply:
also if there is a way to move the user off the wall that would also help with the visual

Reply:
The code to do that is in the source code (ie. Manhattan distance) but from the tests I ran, the diagonal comes up on top for now